src/main/java/org/junit/experimental/results/ResultMatchers.java

```diff
@@ -34,19 +36,22 @@ public ResultMatchers() {
      * Matches if there are {@code count} failures
      */
     public static Matcher<PrintableResult> failureCountIs(final int count) {
-        return new TypeSafeMatcher<PrintableResult>() {
+        return new FailureCountMatcher(equalTo(count)) {
+            @Override
             public void describeTo(Description description) {
                 description.appendText("has " + count + " failures");
             }
-
-            @Override
-            public boolean matchesSafely(PrintableResult item) {
-                return item.failureCount() == count;
-            }
         };
     }
 
     /**
+     * Matches if the number of failures matches {@code countMatcher}
+     */
+    public static Matcher<PrintableResult> failureCountIs(Matcher<? super Integer> countMatcher) {
```

Comment:
Should this be named `failureCountMatches()`? 

Also, please add `@since 4.13` to the Javadoc